Framework/Crystal/test/FindSXPeaksHelperTest.h

```diff
@@ -0,0 +1,394 @@
+        "SXPeak: Should not construct with a zero size specral list",
+        SXPeak(0.001, 0.02, intensity, spectra, 0, spectrumInfo),
+        std::invalid_argument);
+  }
+
+  void testSXPeakGetters() {
+    auto workspace =
+        WorkspaceCreationHelper::create2DWorkspaceWithFullInstrument(10, 10);
+    const auto &spectrumInfo = workspace->spectrumInfo();
+    double intensity = 1;
+    std::vector<int> spectra(1, 1);
+    SXPeak peak(0.001, 0.02, intensity, spectra, 1, spectrumInfo);
+
+    TSM_ASSERT_EQUALS("Intensity getter is not wired-up correctly", 1,
+                      peak.getIntensity());
+    TSM_ASSERT_EQUALS("Detector Id getter is not wired-up correctly", 2,
+                      peak.getDetectorId());
+    // QSpace is also a getter, but is tested more thouroughly below.
+  }
+
+  /* ------------------------------------------------------------------------------------------
+   * Background Strategy
+   * ------------------------------------------------------------------------------------------
+   */
+  void testThatAbsoluteBackgroundPerformsRightComparison() {
+    // GIVEN
+    auto workspace = WorkspaceCreationHelper::create1DWorkspaceConstant(
+        10 /*size*/, 1.5 /*value*/, 1. /*error*/, true /*isHisto*/);
+    const auto y = workspace->y(0);
+
+    // WHEN
+    auto backgroundStrategy = AbsoluteBackgroundStrategy(2.);
+
+    // THEN
+    TSM_ASSERT("The intensity should be below the background",
+               backgroundStrategy.isBelowBackground(1., y));
+    TSM_ASSERT("The intensity should be above the background",
+               !backgroundStrategy.isBelowBackground(2., y));
+  }
+
+  void testThatPerSpectrumBackgroundStrategyPerformsRightComparison() {
+    // GIVEN
+    auto workspace = WorkspaceCreationHelper::create1DWorkspaceConstant(
+        10 /*size*/, 1.5 /*value*/, 1. /*error*/, true /*isHisto*/);
+    const auto y = workspace->y(0);
+
+    // WHEN
+    auto backgroundStrategy = PerSpectrumBackgroundStrategy(1.);
+
+    // THEN
+    TSM_ASSERT("The intensity should be below the background",
+               backgroundStrategy.isBelowBackground(1., y));
+    TSM_ASSERT("The intensity should be above the background",
+               !backgroundStrategy.isBelowBackground(2., y));
+  }
+
+  /* ------------------------------------------------------------------------------------------
+   * Peak Finding strategy
+   * ------------------------------------------------------------------------------------------
+   */
+  void testThatFindsStrongestPeakWhenPerSpectrumBackgroundStrategyIsUsed() {
+    auto backgroundStrategy =
+        Mantid::Kernel::make_unique<PerSpectrumBackgroundStrategy>(1.);
+    doRunStrongestPeakTest(backgroundStrategy.get());
+  }
+
+  void testThatFindsStrongestPeakWhenAbsoluteBackgroundStrategyIsUsed() {
+    auto backgroundStrategy =
+        Mantid::Kernel::make_unique<AbsoluteBackgroundStrategy>(3.);
+    doRunStrongestPeakTest(backgroundStrategy.get());
```

Comment:
Not sure if it is obsolete. In principle `StrongestPeakOnly` and `AbsoluteBackground` can be mixed if we want to do it (hence the whole strategy approach). The test documents that this is a valid combination if we should ever allow this combination. The reason we don't make it available now on the algorithm, is that users tend to dislike to many options.